stock_picking_portal/controllers/portal.py

```diff
@@ -0,0 +1,273 @@
+    )
+    def portal_my_stock_operations(self, **kwargs):
+        values = self._prepare_stock_operations_portal_rendering_values(**kwargs)
+        request.session["my_operation_history"] = values["stock_operation_ids"].ids[
+            :100
+        ]
+        return request.render("stock_picking_portal.portal_my_stock_operations", values)
+
+    def _get_stock_operations_searchbar_sortings(self):
+        return {
+            "date": {"label": _("Order Date"), "order": "scheduled_date desc"},
+            "name": {"label": _("Reference"), "order": "name"},
+            "state": {"label": _("State"), "order": "state"},
+        }
+
+    def _get_stock_operations_searchbar_filters(self):
+        return {
+            "all": {
+                "label": _("All"),
+                "domain": [("picking_type_id.code", "in", ("outgoing", "incoming"))],
+            },
+            "outgoing": {
+                "label": _("Delivery"),
+                "domain": [("picking_type_id.code", "=", "outgoing")],
+            },
+            "incoming": {
+                "label": _("Receipt"),
+                "domain": [("picking_type_id.code", "=", "incoming")],
+            },
+        }
+
+    def _prepare_stock_operations_portal_rendering_values(
+        self,
+        page=1,
+        date_begin=None,
+        date_end=None,
+        sortby=None,
+        filterby=None,
+        **kwargs
+    ):
+        partner = request.env.user.partner_id
+        StockPicking = request.env["stock.picking"]
+        url = "/my/stock_operations"
+        domain = self._get_prepared_operation_domain(partner)
+        if not sortby:
+            sortby = "date"
+        searchbar_filters = self._get_stock_operations_searchbar_filters()
+        # default filter by value
+        if not filterby:
+            filterby = "all"
+        domain += searchbar_filters[filterby]["domain"]
+        values = self._prepare_portal_layout_values()
+        searchbar_sortings = self._get_stock_operations_searchbar_sortings()
+        sort_order = searchbar_sortings[sortby]["order"]
+
+        if date_begin and date_end:
+            domain += [
+                ("scheduled_date", ">", date_begin),
+                ("scheduled_date", "<=", date_end),
+            ]
+        pager_values = portal_pager(
+            url=url,
+            total=StockPicking.search_count(domain),
+            page=page,
+            step=self._items_per_page,
+            url_args={"date_begin": date_begin, "date_end": date_end, "sortby": sortby},
+        )
+        operations = StockPicking.search(
+            domain,
+            order=sort_order,
+            limit=self._items_per_page,
+            offset=pager_values["offset"],
+        )
+        for item in operations:
+            item._portal_ensure_token()
+        values.update(
+            {
+                "date": date_begin,
+                "stock_operation_ids": operations if operations else StockPicking,
```

Comment:
For what do you wrote this condition?
If `operations` is empty then `operations` will equal to `StockPicking` just empty recordset.